utilities/rpp-unittests/OCL/BatchPD_ocl.cpp

```diff
@@ -0,0 +1,729 @@
+    {
+        affine_array[i] = 1.0;
+        affine_array[i + 1] = 1.5;
+        affine_array[i + 2] = 2.0;
+        affine_array[i + 3] = 2.0;
+        affine_array[i + 4] = 2.5;
+        affine_array[i + 5] = 3.0;
+    }
+
+    Rpp32u x1[images];
+    Rpp32u x2[images];
+    Rpp32u y1[images];
+    Rpp32u y2[images];
+
+    for (i = 0; i < images; i++)
+    {
+        x1[i] = 100;
+        x2[i] = 300;
+        y1[i] = 200;
+        y2[i] = 350;
+    }
+    Rpp32u xRoiBegin[images];
+    Rpp32u xRoiEnd[images];
+    Rpp32u yRoiBegin[images];
+    Rpp32u yRoiEnd[images];
+    Rpp32u mirrorFlag[images];
+
+    for (i = 0; i < noOfImages; i++)
+    {
+        xRoiBegin[i] = 50;
+        yRoiBegin[i] = 50;
+        xRoiEnd[i] = 200;
+        yRoiEnd[i] = 200;
+        mirrorFlag[i] = 0;
+        //dstSize[i].height = yRoiEnd[i];
+        //dstSize[i].width =  xRoiEnd[i];
+    }
+    Rpp32u crop_pos_x[images];
+    Rpp32u crop_pos_y[images];
+    for (i = 0; i < images; i++)
+    {
+        crop_pos_x[i] = 100;
+        crop_pos_y[i] = 100;
+    }
+    Rpp32f hueShift[images];
+    Rpp32f saturationFactor[images];
+    for (i = 0; i < images; i++)
+    {
+
+        hueShift[i] = 10;
+        saturationFactor[i] = 5;
+    }
+    Rpp8u minThreshold[images];
+    Rpp8u maxThreshold[images];
+    for (i = 0; i < images; i++)
+    {
+        minThreshold[i] = 10;
+        maxThreshold[i] = 30;
+    }
+    Rpp32u numOfPixels[images];
+    Rpp32u gaussianKernelSize[images];
+    Rpp32f kValue[images];
+    Rpp8u threshold[images];
+    Rpp32f threshold1[images];
+    Rpp32u nonmaxKernelSize[images];
+    for (i = 0; i < images; i++)
+    {
+        numOfPixels[i] = 4;
+        gaussianKernelSize[i] = 7;
+        kValue[i] = 1;
+        threshold[i] = 15;
+        threshold1[i] = 10;
+        nonmaxKernelSize[i] = 5;
+    }
+    Rpp32u sobelType[images];
+    for (i = 0; i < images; i++)
+    {
+        sobelType[i] = 1;
+    }
+    Rpp8u min[images];
+    Rpp8u max[images];
+    for (i = 0; i < images; i++)
+    {
+        min[i] = 10;
+        max[i] = 30;
+    }
+    const int size_perspective = images * 9;
+    Rpp32f perspective[64 * 9];
+    for (i = 0; i < images; i++)
+    {
+        perspective[0 + i * 9] = 1;
+        perspective[1 + i * 9] = 0.5;
+        perspective[2 + i * 9] = 0;
+        perspective[3 + i * 9] = 0.5;
+        perspective[4 + i * 9] = 1;
+        perspective[5 + i * 9] = 0;
+        perspective[6 + i * 9] = 1;
+        perspective[7 + i * 9] = 0;
+        perspective[8 + i * 9] = 0.5;
+    }
+    Rpp32f percentage[images];
+    for (i = 0; i < images; i++)
+    {
+        percentage[i] = 100;
+    }
+
+    cl_mem d_input, d_input_second, d_output;
+    cl_platform_id platform_id;
+    cl_device_id device_id;
+    cl_context theContext;
+    cl_command_queue theQueue;
+    cl_int err;
+    err = clGetPlatformIDs(1, &platform_id, NULL);
+    err = clGetDeviceIDs(platform_id, CL_DEVICE_TYPE_GPU, 1, &device_id, NULL);
```

Comment:
Codacy found an issue: [Variable 'err' is reassigned a value before the old one has been used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/rpp/pullRequest?prid=5976402)